apps/cmp.c

```diff
@@ -3186,8 +3186,7 @@ int cmp_main(int argc, char **argv)
     X509_STORE_free(OSSL_CMP_CTX_get_certConf_cb_arg(cmp_ctx));
     OSSL_CMP_CTX_free(cmp_ctx);
     X509_VERIFY_PARAM_free(vpm);
-    if (engine != NULL) /* workaround for Coverity false positive */
-        release_engine(engine);
+    release_engine(engine);
```

Comment:
It's correct to remove this guard because issue #12277 needs to be solved differently,
as I note here: https://github.com/openssl/openssl/pull/12231#issuecomment-650032345

Reply:
BTW, is that issue #12277 maybe already solved?

At least it does not look like that this PR contains a fix for `OPENSSL_free(e)` being called prematurely in `cyrpto/engine/eng_lib.c`, and I cannot find any more the corresponding Coverity issue 1463570.

Reply:
It's fine to remove the guard since it is done one level down:
``` C
void release_engine(ENGINE *e)
{
#ifndef OPENSSL_NO_ENGINE
    if (e != NULL)
        /* Free our "structural" reference. */
        ENGINE_free(e);
#endif
}
```

Reply:
See #12277 for comment about why its a false positive.

Reply:
> It's fine to remove the guard since it is done one level down:
> 
> ```c
> void release_engine(ENGINE *e)
> {
> #ifndef OPENSSL_NO_ENGINE
>     if (e != NULL)
>         /* Free our "structural" reference. */
>         ENGINE_free(e);
> #endif
> }
> ```

Actually also there the guard can be removed,
since for any (regular) `*_free()` functions calling with a `NULL` arg is a no-op.

Reply:
Good catch.  Fixed now.